sameproject/sdk/same/dataset.py

```diff
@@ -0,0 +1,46 @@
+import yaml
+import os
+from urllib.parse import urlparse
+import pandas as pd
+
+def dataset(name, same_file='same.yaml'):
+    try:
+      env_var=os.environ['SAME_ENV'] if os.environ['SAME_ENV']!="" else "default"
+    except:
+      env_var="default"
+
+
+    with open(same_file, 'r') as file:
+        same_variables = yaml.safe_load(file)
+
+
+    urlpar = urlparse(same_variables['datasets'][name][env_var])
+    
+    url = 'https://gateway.ipfs.io/ipfs/'+urlpar.netloc+urlpar.path if urlpar.scheme=='ipfs' else same_variables['datasets'][name][env_var]
+    filename, file_extension = os.path.splitext(url)
+    
+    extensions = {
+      '.csv': 'read_csv',
+      '.dta': 'read_stata',
+      '.feather': 'read_feather',
+      '.html': 'read_html',
+      '.json': 'read_json',
+      '.orc': 'read_orc',
+      '.parquet': 'read_parquet',
+      '.pickle': 'read_pickle',
+      '.sas': 'read_sas',
+      '.sav': 'read_spss',
+      '.sql': 'read_sql',
+      '.txt': 'read_fwf',
+      '.xml': 'read_xml',
+      ('.hdf', '.h4', '.hdf4', '.he2', '.h5', '.hdf5', '.he5'): 'read_hdf',
+      ('.xlsx', '.ods'): 'read_excel',
+    }
+    
+    for key,value in list(extensions.items()):
+      if type(key)==tuple:
+        for ext in key:
+          extensions[ext]=value
```

Comment:
won't this just keep overwriting the one key with this value?

this whole for loop is a bit tricky to understand. wouldn't it be easier just to split out the extensions dict manually like:
```
    extensions = {
      '.csv': 'read_csv',
      '.dta': 'read_stata',
      '.feather': 'read_feather',
      '.html': 'read_html',
      '.json': 'read_json',
      '.orc': 'read_orc',
      '.parquet': 'read_parquet',
      '.pickle': 'read_pickle',
      '.sas': 'read_sas',
      '.sav': 'read_spss',
      '.sql': 'read_sql',
      '.txt': 'read_fwf',
      '.xml': 'read_xml',
      '.hdf': 'read_hdf',
      '.h4': 'read_hdf',
      '.hdf4': 'read_hdf', # and so on...
    }
```
then you can get rid of this confusing for loop

Reply:
I kept this way because it is temporary and maps the keys to values, In the long term we use pd.io.api and also get extensions that each function supports from pandas, and this way is better to keep track of extensions, In the end both are the same thing

Reply:
but surely the current implementation doesn't work... like i said, won't this assignment keep overwriting the one key with multiple values? so only the last value in the tuple will end up in the dict

Reply:
oh yeah, ok, i managed to convince myself that it works